app/services/versioned_files_service/paths.rb

```diff
@@ -0,0 +1,88 @@
+class VersionedFilesService
+  # Support for paths.
+  class Paths
```

Comment:
I don't like how this mixes a bunch of different kinds of paths.  I think the AwflPaths could be in on class, but the transfer path and purl paths should be in separate classes.

Reply:
Agree. Done.